ntp-proto/src/packet/mod.rs

```diff
@@ -235,6 +235,12 @@ impl NtpHeaderV3V4 {
         recv_timestamp: NtpTimestamp,
         clock: &C,
     ) -> Self {
+        let reference_timestamp = match input.reference_timestamp {
+            #[cfg(feature = "ntpv5")]
+            v5::UPGRADE_TIMESTAMP => v5::UPGRADE_TIMESTAMP,
+            _ => Default::default(),
+        };
```

Comment:
I think we should only response with the upgrade timestamp if there is also a draft id extension field for the draft we implement. Otherwise we will confuse future implementations saying that we support ntpv5 when we don't support theirs.

Reply:
I can add that, but it is not mentioned in the draft, so clients would usually not send a draft id extension by default I guess...

Reply:
I'll write a proposal to add this more explicitly to the next draft. For right now, since we are the first implementation (other than a few demo implementations that don't do time sync) I think we can just require this one-sided.

Reply:
Done!